src/components/MessagesList/MessagesList.vue

```diff
@@ -351,18 +350,16 @@ export default {
 		 * @returns {string} The last known message id.
 		 */
 		getLastKnownMessageId() {
-			for (let i = 1; i < this.messagesList.length; i++) {
-				if (this.messagesList[this.messagesList.length - i]) {
-					const id = this.messagesList[this.messagesList.length - i].id.toString()
-					// Ignore temporary messages
-					if (!id.startsWith('temp-')) {
-						return id
-					}
-				}
+			// Reverse a copy of the messages array
+			const reversedMessages = this.messagesList.slice().reverse()
+			// Get the id of the last non-temporary message
+			for (const message of reversedMessages) {
+				const id = message.id.toString()
+				if (!id.startsWith('temp-')) {
+					return id
+				} else return '0'
```

Comment:
This should be illegal...
Always use `{}`

Also you return here always on the first item. So if there is a temporary message you return with the 0 instead of looping to the next message.